frappe/public/js/frappe/form/controls/autocomplete.js

```diff
@@ -19,25 +19,61 @@ frappe.ui.form.ControlAutocomplete = frappe.ui.form.ControlData.extend({
 	},
 
 	get_awesomplete_settings() {
+		var me = this;
 		return {
 			minChars: 0,
 			maxItems: 99,
 			autoFirst: true,
 			list: this.get_data(),
+			data: function(item) {
+				if(!(item instanceof Object)) {
+					var d = {"value": item};
+					item = d;
+				}
+
+				return {
+					label: item.label || item.value,
+					value: item.value
+				};
+			},
+			filter: function() {
+				return true;
+			},
+			item: function(item) {
+				var d = this.get_item(item.value);
+				if(!d) {
+					d = item;
+				} else {
+					d.label = d.value;
+				}
+
+				var _label = (me.translate_values) ? __(d.label) : d.label;
+				var html = "<strong>" + _label + "</strong>";
+				if(d.description && d.value!==d.description) {
+					html += '<br><span class="small">' + __(d.description) + '</span>';
+				}
+
+				return $('<li></li>')
+					.data('item.autocomplete', d)
+					.prop('aria-selected', 'false')
+					.html('<a><p>' + html + '</p></a>')
+					.get(0);
+			},
 			sort: () => {
 				return 0;
 			}
 		};
 	},
 
 	setup_awesomplete() {
+		var me = this;
 		this.awesomplete = new Awesomplete(this.input, this.get_awesomplete_settings());
 
 		$(this.input_area).find('.awesomplete ul').css('min-width', '100%');
 
-		this.$input.on('input', () => {
-			this.awesomplete.list = this.get_data();
-		});
+		this.$input.on('input', frappe.utils.debounce(function() {
```

Comment:
You can still use an arrow function for the debounced callback to get rid of the `me` :)